src/ExpressionEvaluator/Core/Test/ResultProvider/Debugger/Engine/DkmClrValue.cs

```diff
@@ -310,7 +289,7 @@ public string EvaluateDebuggerDisplayString(string formatString)
                 throw new ArgumentException(string.Format("Unmatched open brace in '{0}'", formatString));
             }
 
-            return pooled.ToStringAndFree();
+            completionRoutine(new DkmEvaluateDebuggerDisplayStringAsyncResult(pooled.ToStringAndFree()));
```

Comment:
Something like this seems like a good idea, but it now seems like we'd want all the work to be scheduled on the same thread (to mimic the "request" thread in the Concord dispatcher).

I'm going to hold off on this for now...